File: DI/DI-Dagger-Java/src/main/java/model/persona/repositories/PersonaRepository.java
package model.persona.repositories;

import model.persona.model.Persona;
import model.persona.services.IPersonaStorage;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.UUID;

/**
 * El repositorio almacena los datos de las personas.
 */
@Singleton
public class PersonaRepository implements IPersonaRepository {
    private final IPersonaStorage storage;
    private final String id = UUID.randomUUID().toString();

    @Inject
    public PersonaRepository(IPersonaStorage storage) {
        this.storage = storage;
    }

    @Override
    public Persona save(Persona entity) {
        System.out.println("PersonaRepository.save() -->" + entity);
        return storage.save(entity);
    }

    @Override
    public String toString() {
        return "PersonaRepository{" +
                "id='" + id + '\'' +
                ", storage=" + storage +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/personas/repositories/PersonaRepository.kt
package models.personas.repositories

import models.personas.model.Persona
import models.personas.services.IPersonaStorage
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton


/**
 * El repositorio almacena los datos de las personas.
 */
@Singleton
class PersonaRepository
@Inject constructor(
    private val storage: IPersonaStorage
) : IPersonaRepository {
    private val id = UUID.randomUUID().toString()

    override fun save(entity: Persona): Persona {
        println("PersonaRepository.save() -->$entity")
        return storage.save(entity)
    }

    override fun toString() = "PersonaRepository(storage=$storage, id='$id')"

}
